openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -591,14 +591,27 @@ OpenAssessment.ResponseView.prototype = {
         this.filesType = uploadType;
         this.filesUploaded = false;
 
-        var ext = null;
-        var fileType = null;
         var errorCheckerTriggered = false;
 
-        for (var i = 0; i < files.length; i++) {
-            ext = files[i].name.split('.').pop().toLowerCase();
-            fileType = files[i].type;
+        var isUploadSupported = function(file, uploadType) {
+            var ext = file.name.split('.').pop().toLowerCase();
+            var fileType = file.type;
 
+            // Check upload type/extension matches allowed types
+            if (uploadType === 'image' && this.data.ALLOWED_IMAGE_MIME_TYPES.indexOf(fileType) === -1) {
```

Comment:
Could there ever be a case where a file is of the correct mime type but of a different file type and we allow it although it's not allowed by the file type list?

Reply:
I guess this would mostly be covered by the input not allowing you to select things of non-supported extensions

Reply:
**TL;DR -** yes this could happen but I wouldn't expect it to cause any issues.

Right now, presets both in the frontend and backend are validated based on mime type and not file type. Only in the case of custom extensions do we block based on a bad file type. For presets, the shown extension list is more of a visual aide and hint to the file select menu than an explicit blocker. The file select menu in the browsers will automatically restrict you to the allowed filetypes but you could coerce it to a bizarre file type that still matches the mime type and I would expect it to go through w/out a problem.